src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapperJsonConverter.cs

```diff
@@ -0,0 +1,308 @@
+            {
+                return false;
+            }
+
+            return typeof(ISelectExpandWrapper).IsAssignableFrom(typeToConvert);
+        }
+
+        /// <inheritdoc/>
+        public override JsonConverter CreateConverter(
+            Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            if (typeToConvert == null || !typeToConvert.IsGenericType)
+            {
+                return null;
+            }
+
+            Type genericType = typeToConvert.GetGenericTypeDefinition();
+            Type type = typeToConvert.GetGenericArguments()[0];
+
+            if (genericType == typeof(SelectExpandBinder.SelectSome<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectSomeOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+            else if (genericType == typeof(SelectExpandBinder.SelectSomeAndInheritance<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectSomeAndInheritanceOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+            else if (genericType == typeof(SelectExpandBinder.SelectAll<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectAllOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+            else if (genericType == typeof(SelectExpandBinder.SelectAllAndExpand<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectAllAndExpandOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+            else if (genericType == typeof(SelectExpandWrapper<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectExpandWrapperOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+
+            return null;
+        }
+
+        private class JsonPropertyNameMapper : IPropertyMapper
+        {
+            private IEdmModel _model;
+            private IEdmStructuredType _type;
+
+            public JsonPropertyNameMapper(IEdmModel model, IEdmStructuredType type)
+            {
+                _model = model;
+                _type = type;
+            }
+
+            public string MapProperty(string propertyName)
+            {
+                IEdmProperty property = _type.Properties().Single(s => s.Name == propertyName);
+                PropertyInfo propertyInfo = GetPropertyInfo(property);
+
+                Newtonsoft.Json.JsonPropertyAttribute jsonPropertyAttribute;
+                JsonPropertyNameAttribute jsonPropertyNameAttribute;
+
+                // NOTE: Preceding .NET Core 3.1 Newtonsoft was the default serialization library
+                // We check the Newtonsoft JsonPropertyAttribute first to prevent breaking changes
```

Comment:
👍
I was wondering why we're referencing Newtonsoft.Json in System.Text.Json converter. Thanks for clearing that up.